### ISSUE TITLE
Don't write json in the middle of responses

### ISSUE THREAD

**Issue by carlhoerberg** [cloudamqp/avalanchemq#341]
This will write json directly to the response, which already holds data, making the resulting json invalid: 

https://github.com/cloudamqp/avalanchemq/blob/183a65cf7ed6c32094bc54487f551266759a3b47/src/avalanchemq/http/controller/prometheus.cr#L174-L176

**Comment by snichme** [comment]
I've seen this before, I think we should rewind the response `IO` on any error and then write the error. Not sure if we could make a general error handler or just do it in all `rescue` blocks

**Comment by carlhoerberg** [comment]
I don't think you can rewind `context.response`, it's a socket afaik. I truncate the result set (and log warn) instead in #343
